### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by MickaelAlliel**

### CODE REVIEW

File: src/services/utils.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const cleanString = (str) => {
```
Comment (MickaelAlliel) [inline src/services/utils.js]: This function is kept for bug hunting purposes only

File: src/services/getTodosByFilter.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+const getTodos = async ({ fetchFunc, labelText, filter }) => {
+  const todos = await fetchFunc();
+
+  return {
+    shownTodos: todos,
+    countLabel: `${todos.length} ${labelText}`,
```
Comment (iglizer) [inline src/services/getTodosByFilter.js]: countText / summary?

File: src/services/getTodosByFilter.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const fetchActiveTodos = () => global.Store.find({ completed: false });
+
+  return getTodos({
+    fetchFunc: fetchActiveTodos,
+    labelText: 'Active Tasks',
+    filter: 'active',
+  });
+};
+
+const todoByFilter = {
```
Comment (iglizer) [inline src/services/getTodosByFilter.js]: change name to fetchByFilter

Comment (iglizer) [reply src/services/getTodosByFilter.js]: change name to fetchTodosByFilter

File: src/services/getTodosByFilter.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  completed: getCompleted,
+  active: getActive,
+  all: getAll,
+};
+
+const getTodosByFilter = async (filter) => {
+  const todosByFilter = todoByFilter[filter] || todoByFilter.all;
```
Comment (iglizer) [inline src/services/getTodosByFilter.js]: todosByFilter  => fetchTodos

File: src/services/todos.js [hunk 5]
```diff
@@ -1,66 +1,116 @@
-// as we discussed, "home" is a frontend decision, so no need to mention it here.
-// also the word "data" does not really contribute
-///  getHomeCompletedData => getCompleted etc.
+/* eslint prefer-destructuring: 0 */
+const utils = require('./utils');
 
+const actionResult = (ok, message, errorCode = null, errorMessage = null) => ({
```
Comment (iglizer) [inline src/services/todos.js]: 👍

**Pull request merged**